### PR TITLE
Sync wrappers over async methods replaced by regular methods in facades

### PR DESCRIPTION
**Pull request by Leeemons**
This PR solves issues with _sync_ wrappers overs _async_ methods introduced in PR #31.

### CODE REVIEW

**Comment by luboshl** [comment]
@Leeemons, please add some description to this PR.

File: src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/CrudFacadeBase.cs [hunk 1]
```diff
@@ -201,9 +262,29 @@ protected virtual void PopulateDetailToEntity(TDetailDTO detail, TEntity entity)
         /// Validates that the entity detail can be displayed by the user. If the user does not have permissions, the method should throw an exception.
         /// </summary>
         /// <param name="entity"></param>
-        protected virtual Task ValidateReadPermissions(TEntity entity)
+        protected virtual Task ValidateReadPermissionsAsync(TEntity entity)
```
Comment (luboshl) [inline src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/CrudFacadeBase.cs]: To be consistent (async method after sync), move this method after `ValidateReadPermissions(TEntity entity)`.

File: src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/FilteredCrudFacadeBase.cs [hunk 2]
```diff
@@ -41,10 +47,9 @@ public virtual IEnumerable<TListDTO> GetList(TFilterDTO filter, Action<IFiltered
             using (UnitOfWorkProvider.Create())
             {
                 var query = QueryFactory();
-                query.Filter = filter;
```
Comment (luboshl) [inline src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/FilteredCrudFacadeBase.cs]: Why have you removed the filter?

**Review by luboshl — CHANGES_REQUESTED** [review]
1. Add `CancellationToken` argument with the default value to all added async methods and use the value whenever possible.

2. `TemporalRelationshipCrudFacade` - this class was modified in PR #31, this PR does not fix the issues:
- `GetList` has not been fixed, still acts as a wrapper over async method
- `GetDetail` - DTTO
- `Save` - DTTO
- `Delete` - DTTO
- missing `GetRelationshipCollection` method
- `ValidateReadPermissions` - method was changed to async, but not renamed to `ValidateReadPermissionsAsync`. Missing sync version of the method.
- `HasReadPermissions` - DTTO
- `ValidateModifyPermissions` - DTTO
- `HasModifyPermissions` - DTTO

**Comment by Leeemons** [comment]
@luboshl I tried fixed all issues you reported

**Review by luboshl — COMMENTED** [review]
Please move the `CancellationToken` argument to be the last one and add the default value.

**Review by luboshl — CHANGES_REQUESTED** [review]
Please move the `CancellationToken` argument to be the last one and add the default value.

File: src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs [hunk 3]
```diff
@@ -254,24 +366,49 @@ protected virtual void EndEntityValidityPeriod(TRelationshipEntity entity, DateT
...
                 throw new UnauthorizedAccessException();
             }
         }
 
-        protected virtual Task<bool> HasReadPermissions(TParentEntity entity) => Task.FromResult(true);
+        protected virtual bool HasModifyPermissions(TParentEntity entity) => true;
+
 
-        protected virtual async Task ValidateModifyPermissions(TParentEntity entity)
+        protected virtual async Task ValidateReadPermissionsAsync(CancellationToken cancellationToken, TParentEntity entity)
```
Comment (luboshl) [inline src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs]: Move this method just after its corresponding sync version to be together.

File: src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs [hunk 4]
```diff
@@ -254,24 +366,49 @@ protected virtual void EndEntityValidityPeriod(TRelationshipEntity entity, DateT
...
-            if (!await HasModifyPermissions(entity))
+            cancellationToken.ThrowIfCancellationRequested();
+
+            if (!await HasReadPermissionsAsync(entity))
+            {
+                throw new UnauthorizedAccessException();
+            }
+        }
+
+        protected virtual Task<bool> HasReadPermissionsAsync(TParentEntity entity) => Task.FromResult(true);
```
Comment (luboshl) [inline src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs]: Move this method just after its corresponding sync version to be together.

File: src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs [hunk 5]
```diff
@@ -254,24 +366,49 @@ protected virtual void EndEntityValidityPeriod(TRelationshipEntity entity, DateT
...
+
+        protected virtual async Task ValidateModifyPermissionsAsync(CancellationToken cancellationToken, TParentEntity entity)
```
Comment (luboshl) [inline src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs]: Move this method just after its corresponding sync version to be together.

File: src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs [hunk 6]
```diff
@@ -254,24 +366,49 @@ protected virtual void EndEntityValidityPeriod(TRelationshipEntity entity, DateT
...
+            cancellationToken.ThrowIfCancellationRequested();
+
+            if (!await HasModifyPermissionsAsync(entity))
             {
                 throw new UnauthorizedAccessException();
             }
         }
 
-        protected virtual Task<bool> HasModifyPermissions(TParentEntity entity) => Task.FromResult(true);
+        protected virtual Task<bool> HasModifyPermissionsAsync(TParentEntity entity) => Task.FromResult(true);
```
Comment (luboshl) [inline src/Infrastructure/Riganti.Utils.Infrastructure.Services/Facades/TemporalRelationshipCrudFacade.cs]: Move this method just after its corresponding sync version to be together.

**Review by luboshl — CHANGES_REQUESTED** [review]
Still some (but small) issues.

**Comment by luboshl** [comment]
As we discussed personally, `CancellationToken` really should be the last argument of the method. The situation is different in the Repository, because there are methods with the `params` argument and it has to be the last, there is no other option. In the future we could refactore these methods not to use `params` but `ICollection` as an argument for example, then the `CancellationToken` can also be the last one.

**Review by luboshl — APPROVED** [review]

**Pull request merged**